mllib/src/main/scala/org/apache/spark/ml/param/params.scala

```diff
@@ -65,38 +90,129 @@ class Param[T] (val parent: Params, val name: String, val doc: String) extends S
   }
 }
 
+/**
+ * Factory methods for common validation functions for [[Param.isValid]].
+ * The numerical methods only support Int, Long, Float, and Double.
+ */
+object ParamValidate {
```

Comment:
We might need `sealed abstract class ParamValidate` to make it show up correctly in Java. For the name, I recommend `ParamValidators` instead of `ParamValidate`.